src/tridiagonal.rs

```diff
@@ -0,0 +1,805 @@
+//! Vectors as a Tridiagonal matrix
+//! &
+//! Methods for tridiagonal matrices
+
+use std::ops::{Index, IndexMut};
+
+use cauchy::Scalar;
+use ndarray::*;
+use num_traits::One;
+
+use super::convert::*;
+use super::error::*;
+use super::lapack::*;
+use super::layout::*;
+
+/// Represents a tridiagonal matrix as 3 one-dimensional vectors.
+/// This struct also holds the layout of the raw matrix.
+#[derive(Clone, PartialEq)]
+pub struct Tridiagonal<A: Scalar> {
+    /// layout of raw matrix
+    pub l: MatrixLayout,
+    /// (n-1) sub-diagonal elements of matrix.
+    pub dl: Vec<A>,
+    /// (n) diagonal elements of matrix.
+    pub d: Vec<A>,
+    /// (n-1) super-diagonal elements of matrix.
+    pub du: Vec<A>,
+}
+
+pub trait TridiagIndex {
+    fn to_tuple(&self) -> (i32, i32);
+}
+impl TridiagIndex for [Ix; 2] {
+    fn to_tuple(&self) -> (i32, i32) {
+        (self[0] as i32, self[1] as i32)
+    }
+}
```

Comment:
I think [`ndarray::NdIndex<Ix2>`](https://docs.rs/ndarray/0.13.1/ndarray/trait.NdIndex.html) could be better for it, but it does not enough iterface for implementing this PR, e.g. `Into<Ix2>`. I keep this code as it is.